### ISSUE TITLE
[class java.lang.String cannot be cast to class java.lang.Number (java.lang.String and java.lang.Number are in module java.base of loader 'bootstrap')]

### ISSUE THREAD

**Issue by JoShMiQueL** [viclew1/VLDofusBot#30]
### **When I try to perform a treasure hunt I get this error**

![image](https://github.com/viclew1/VLDofusBot/assets/26714939/fc6ed528-aa75-43fa-b815-9f646bb902ae)

**Comment by JoShMiQueL** [comment]
One more thing, I am manually adding POI's, do I really have to do this with all of them? wtf Is there no way you can fix this?

**Comment by viclew1** [comment]
I can't reproduce this error, have you changed some code or manually modified config files ?
Also, please keep it polite. I don't think I'm working for you.
You have to do it with every POI gfx ids. Once you register every gfx of a POI, the bot will find it no matter the map. I could of course find a way to make it better, but I don't want to spend time updating the treasure hunt. It is not a feature I find valuable and I don't use it.

You can put this in your hint_gfx_ids_by_label though if you have very little gfx ids registered, it will be a good start :
`{"Canard en plastique":[70551,70552,31149],"Crâne dans un trou":[69460,69465,69466],"Rose noire":[63745,63746,52307,70571],"Chaussette à pois":[400319],"Bouton de couture":[70553,70554],"Flèche dans une pomme":[400322],"Soupe de bananagrumes":[69622],"Palmier peint à rayures":[69640],"Balançoire macabre":[400313],"Ceinture cloutée":[55959,70555,70556],"Bougie dans un trou":[69462],"Bannière brâkmarienne déchirée":[400315],"Gravure de Gelax":[69504],"Gravure de crâne":[69445,69477,69510,69416],"Poupée koalak":[45690],"Gravure de Kama":[69488,69478],"Buisson poupée sadida":[400318],"Champignon rayé":[69451,69450],"Bannière bontarienne déchirée":[400314,40031],"Tombe gravée d'un bouclier":[69598],"Plaque gravée d'un symbole égal":[69444],"Ancre dorée":[69528],"Gravure de logo Ankama":[69475,69509,69496],"Plaque gravée d'une Epée":[69476],"Crâne de cristal":[69649],"Blé noir et blanc":[400311],"Tambour à rayures":[69621],"Tombe inondée de sang":[69599],"Rocher à sédimentation verticale":[70535,70536],"Statue koalak":[69617,69616],"Girouette dragodinde":[69458],"Anneau d'or":[70561,51984,70560,70562,70564,36219,70559,70558],"Œil de shushu peint":[69467],"Marionnette":[600074],"Peinture de Dofus":[69417,69418],"Gravure de soleil":[69441,69437],"Échelle cassée":[600072],"Kama peint":[69413,69414],"Gravure de fleur":[69479,69512,69500],"Slip à petit cœur":[52320,36741,69526],"Corne de likrone":[57572],"Gravure de symbole de quête":[69473,69508,69497],"Dolmen":[70533],"Arche naturelle":[45776],"Niche dans une caisse":[70541],"Gravure de Dofus":[69495,69515],"Ballons en forme de cœur":[69604],"Peluche de likrone":[301567],"Tube rempli de tofus":[69527],"Poisson grillé embroché":[70543],"Gravure de tofu":[69487],"Chapeau dé":[69468],"Minouki":[69464],"Gravure d'étoile":[69443,69438],"Boule dorée de marin":[69642],"Gravure de wukin":[69511,69501],"Gravure de flèche":[69474,69481,69499],"Kaliptus à fleurs jaunes":[69614],"Rocher Dofus":[69613],"Carapace de tortue":[70579],"Gravure d'arakne":[69484],"Croix en pierre brisée":[70537],"Dessin dragodinde":[28272,45693,45695],"Paire de lunettes":[70550],"Crâne de likrone dans la glace":[400331],"Kaliptus coupé":[45698,45699],"Gravure de spirale":[69434],"Gravure de clef":[69492,69516],"Plaque gravée d'un crâne":[69445,69477],"Plaque gravée d'un logo Ankama":[69475],"Rose des vents dorée":[69427,69429],"Logo Ankama peint":[69419],"Fer à cheval":[54480,54449],"Tricycle":[38180,38779],"Stèle chacha":[69618],"Gravure de bouftou":[69506],"Rune nimbos":[70569,70570],"Plaque gravée d'une flèche":[69474,69481],"Plaque gravée d'un wukin":[69483],"Tissu à carreaux noué":[69449],"Symbole de quête peint":[69415],"Rocher crâne":[69620],"Dé en glace":[400310],"Crâne de renne":[70539],"Œuf dans un trou":[69461,49032,45689],"Arbre glacé":[600075,69612],"Gravure d'Epée":[69476,69498],"Tombe inondée":[59594,69594,69599],"Plaque gravée d'un fantôme":[400336],"Affiche de carte au trésor":[44786,70547,16643],"Casque à cornes":[70545],"Clef dorée":[69517,69518],"Squelette d'Ouginak pendu":[400335],"Pioche plantée":[70567,70566],"Rocher taillé en arètes de poisson":[69592],"Aiguille à coudre":[70573,653,70574],"Théière à rayures":[69469,69470],"Sève qui s'écoule":[45683],"Palmier à feuilles carrées":[69635],"Gravure d'aile":[69514,69502],"Éolienne à quatre pales":[69624],"Dessin koalak":[28267,45692,28271],"Palmier à pois":[69638],"Menottes":[15440,15433,70540],"Statue wabbit":[51296],"Framboisier":[70572],"Plaque gravée d'une lune":[69442],"Serrure dorée":[69447],"Langue dans un trou":[69463],"Gravure de wabbit":[69493],"Moufles jaunes":[400334],"Plaque gravée d'une étoile":[69443],"Crâne de Roublard":[69456],"Talisman en papier":[69454,67086],"Arbre à moitié coupé":[400312],"Crâne de likrone":[400330],"Fleurs smiley":[69453],"Crâne de Crâ":[69457],"Cadran solaire":[70578],"Gravure de chacha":[69486],"Plaque gravée d'un cœur":[400324],"Étoile jaune peinte":[69411,69412],"Torii cassé":[69629],"Rocher dé":[69615],"Palmier à feuilles déchirées":[69636],"Gravure de lune":[69442,69494,69480,69436],"Dofus en bois":[69530],"Canne à kebab":[400327],"Lanterne au crâne luminescent":[70549],"Tambour papatte":[69626],"Grelot":[52183,1597,1614,1598],"Arbre ensanglanté":[69588],"Plaque gravée d'un œil":[69482],"Kaliptus grignoté":[45700],"Sucre d'orge":[38472],"Moufles rouges":[400333],"Hache brisée":[70546],"Dessin de croix dans un cercle":[69420],"Gravure de papatte":[69472,69507,69490],"Barque coulée":[69643],"Gravure d'œil":[69513,69435,69482,69485],"Corail avec des dents":[69531],"Palmier surchargé de noix de coco":[69637],"Plaque gravée d'un soleil":[69441],"Panneau nonosse":[45762],"Gravure de rose des vents":[69425,69424,69426],"Étoile en papier plié":[400320,400321,400316],"Gravure de croix":[69433,69432],"Gravure de boule de poche":[69431],"Ornement flocon":[38582],"Gravure de Firefoux":[69505],"Lampion bleu":[69623],"Bombe cœur":[400328],"Coquillage à pois":[70534],"Sapin couché":[70538],"Grand coquillage cassé":[69583],"Épouvantail à pipe":[400317],"Gravure de symbole égal":[69444],"Crâ cramé":[69611],"Trace de main en sang":[2098,39477,39478],"Gravure de dragodinde":[69503],"Queue d'Osamodas":[20302],"Bonbon bleu":[400329],"Plaque gravée d'un symbole de quête":[69473],"Cairn":[70548],"Os dans la lave":[70568]}`

**Comment by JoShMiQueL** [comment]
I am sorry for my previous answer, I think I did not express myself correctly.

Regarding the first error, it doesn't happen anymore, I don't know why

Regarding the POI, there are some maps that I can't access to the GFX ID of the object because there are too many together.
![image](https://github.com/viclew1/VLDofusBot/assets/26714939/decf4920-d927-4007-95a3-3bcb2bf455b8)
![image](https://github.com/viclew1/VLDofusBot/assets/26714939/53d4faf6-dfd8-4b72-962e-ad6b6d6f8f84)
Or the POI is in a totally different place and is very confusing.
![image](https://github.com/viclew1/VLDofusBot/assets/26714939/9676a0a2-491c-4afd-b908-e2cb361a7085)

Why do you feel that treasure hunting is not useful? I think it is a very good feature.
> I don't want to spend time updating the treasure hunt. It is not a feature I find valuable and I don't use it.

**Comment by viclew1** [comment]
Yeah I'm aware of the issues (also met them when I filled the file I sent you), I used to refer to an external for hints, but I was dependent of their down times (which happened a lot more than you'd expect). I made this solution to have, once every hint met, a good solution to find them anywhere they are.

> Why do you feel that treasure hunting is not useful? I think it is a very good feature.

Thank you, but I don't find the gain worth the time spent doing the hunts, maybe on other servers than mine roses are still valuable though, I don't know. What I'll probably do however is improve the overlays (every overlay), it's slow, buggy and kinda hard to use atm. I'm not giving you any date on when I plan on doing it.
But until I find a better solution that doesn't take me a lot of time, the system will remain the same.
